### DISCUSSION TITLE
IInputDevice из примера в документации не генерирует EventReceived

### DISCUSSION THREAD

**Discussion by movaxcs** [melanchall/drywetmidi#315] [Q&A] [answered]
Добрый день!

Не срабатывает обработчик EventReceived при нажатии клавиш на пианино. Пианино соединено кабелем, видно для различных приложений в обе стороны (работает и MIDI in, и MIDI out). Но код из примера не хочет работать ни в какую( Пробовал и консольное приложение и WPF (.NET версии 7,8,9)

Прошу помощи, в чем может быть проблема.

```
using Melanchall.DryWetMidi.Multimedia;

namespace ConsoleApp1
{
    internal class Program
    {
        private static IInputDevice _inputDevice;

        static void Main(string[] args)
        {
            _inputDevice = InputDevice.GetByName("Roland Digital Piano");
            _inputDevice.EventReceived += OnEventReceived;
            _inputDevice.StartEventsListening();

            Console.WriteLine("Input device is listening for events. Press any key to exit...");
            Console.ReadKey();

            (_inputDevice as IDisposable)?.Dispose();
        }

        private static void OnEventReceived(object sender, MidiEventReceivedEventArgs e)
        {
            var midiDevice = (MidiDevice)sender;
            Console.WriteLine($"Event received from '{midiDevice.Name}' at {DateTime.Now}: {e.Event}");
        }
    }
}
```

![Screenshot 2024-11-17 031148](https://github.com/user-attachments/assets/a647dad2-1351-48c0-b549-718f485417b8)

**Comment by melanchall** [comment, marked as answer]
Здравствуйте!

1. Не работает и в Debug и в Release (собрать и запустить exe просто)?
2. В других приложениях события приходят?

**Comment by movaxcs** [comment]
![Screenshot 2024-11-17 163028](https://github.com/user-attachments/assets/9bd1ebc1-2cb8-46e1-a395-3e079f16993e)
Debug, все понятно. Большое спасибо! 

Отличная библиотека)

**Reply by melanchall** [reply]
По идее в Debug тоже должно работать. Просто вы встали на брейкпойнте, и выполнение программы, разумеется, встало на паузу на этой точке. Снимите точку останова и всё будет в порядке (я надеюсь).
